### PR TITLE
Integrate unit type for payment history

### PR DESCRIPTION
**Pull request by mohitb35**
Changes
1. Updates type `PaymentHistoryRecord`
2. Disables donor certificate/gift certificate for projects with `unitType = "m2"`
3. Updates text for per unit cost in `AccountRecord` based on `unitType`
4. Updates text for title in `AccountRecord` based on `unitType`, `purpose`, and whether the donation has a gift recipient.

### CODE REVIEW

**Review by Shreyaschorge — APPROVED** [review]
LGTM!

File: src/features/user/Account/components/AccountRecord.tsx [hunk 1]
```diff
@@ -39,27 +28,37 @@ export function RecordHeader({
...
+      case 'trees':
+      case 'conservation':
+        // Sample title 1 => 5 Tree Gift . Yucatan,
+        // Sample title 2 => 2 m² Donation . Sumatra
+        title = `${getFormattedNumber(i18n.language, record.quantity)} ${t(
+          `common:${record.unitType}`,
+          { count: 1 }
+        )} ${
+          record.details.giftRecipient ? t('me:gift') : t('me:donation')
+        } . ${record.details.project}`;
```
Comment (Shreyaschorge) [inline src/features/user/Account/components/AccountRecord.tsx:42]: Should this be a dot? or a bullet point?
I think bullet point will look much better.

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]
@mohitb35 Just a minor feedback, the rest looks good.

File: src/features/user/Account/components/AccountRecord.tsx (hunk 1 above)
Comment (mohitb35) [reply src/features/user/Account/components/AccountRecord.tsx]: Done in fd94e4c

**Review by Shreyaschorge — APPROVED** [review]
LGTM

**Pull request merged**